openedx/core/djangoapps/credit/models.py

```diff
@@ -478,8 +498,11 @@ def update_eligibility(cls, requirements, username, course_key):
                     username=username,
                     course=CreditCourse.objects.get(course_key=course_key),
                 )
+                return True
             except IntegrityError:
-                pass
+                return False
```

Comment:
I think that we need to send a tuple `(is_eligible, eligibility_record_created)` as response for this method since the user is eligible but its `CreditEligibility` record has been created previously and we need to send credit eligibility notification for first time only (when the `CreditEligibility` record is created).